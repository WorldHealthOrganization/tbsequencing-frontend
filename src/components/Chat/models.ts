export interface IUserMessage {
  email: string,
  username: string,
  firstName: string,
  lastName: string
}
