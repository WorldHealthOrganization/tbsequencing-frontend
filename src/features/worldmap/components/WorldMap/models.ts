export type MapType = 'macroRegions' | 'heatmap';
