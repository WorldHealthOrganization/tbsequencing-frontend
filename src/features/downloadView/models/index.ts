import { IDrug } from '../../../services/drugsApi/models';

export interface IExtendedDrug extends IDrug {
  isChecked: boolean;
}
