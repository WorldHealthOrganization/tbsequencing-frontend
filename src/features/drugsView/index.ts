import DrugsView from './componets/DrugsView/DrugsView';

export default DrugsView;
