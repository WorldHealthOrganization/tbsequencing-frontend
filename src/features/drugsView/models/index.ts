export enum DrugStatus {
  Susceptible = 'susceptible',
  Resistant = 'resistant',
  Intermediate = 'intermediate',
  RatioSusceptible = 'ratioSusceptible',
  RatioResistant = 'ratioResistant',
  RatioIntermediate = 'ratioIntermediate',
}

export enum ResistantType {
  Phenotypic = 'Pheno',
  Genotypic = 'Geno',
}

export enum ChartDataType {
  Ratio = 'ratio',
  Count = 'count',
}

export enum Regions {
  ALL = 'ALL_REGIONS',
  AFRICA = 'AFR',
  EUROPE = 'EUR',
  AMERICA = 'AMR',
  EASTERN_MEDITERRANEAN = 'EMR',
  WESTERN_PACIFIC = 'WPR',
  SOUTH_EAST_ASIA = 'SEA',
}
