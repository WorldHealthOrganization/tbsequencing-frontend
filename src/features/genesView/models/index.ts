import { skipToken } from '@reduxjs/toolkit/query';
import React from 'react';
import { IGene, IVariant } from '../../../services/genesApi/models';

export type AutocompleteParams = typeof skipToken | { search?: string };

export interface IGenomeContextWrapperProps {
  data?: IGene[];
  isLoading?: boolean;
  selectedGene?: IGenesOverview;
  minDomain?: number;
  maxDomain?: number;
}

export interface IGenomeContextProps extends IGenomeContextWrapperProps {
  width: number;
}

export interface IVariantLocal extends IVariant {
  pos: number;
}

export interface ISearchResult {
  geneName: string;
  geneDbCrossrefId: number;
  locusTag: string;
}

export interface ISearchCount extends ISearchResult {
  counter: number;
}

export interface ISearchRecent extends ISearchResult {
  counter: string;
  date: Date | string;
}

export interface IGenesOverview extends IGene {
  geneDescription: string;
  geneType: string;
  ncbiId: string;
  proteinLength: number;
}

export interface IGenesOverviewTableData {
  title: string;
  value: string | number | null;
  element?: React.ReactNode;
}

export interface IGenesOverviewResponse {
  genes: IGene;
  genesOverview: IGenesOverview
}
