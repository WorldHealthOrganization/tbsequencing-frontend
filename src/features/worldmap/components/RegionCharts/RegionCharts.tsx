/** @jsxImportSource @emotion/react */
import React from 'react';
import { css } from '@emotion/react';
import { IAggregatedGlobalDrug } from '../../../../services/drugsApi/models';
import DrugsPieChart from '../DrugsPieChart';
import { Regions } from '../../../drugsView/models';
import { IRectDimensions } from '../WorldMap/utils/renderingHelpers';
import { WORLD_MAP_PIE_CHART_SIZE } from '../DrugsPieChart/DrugsPieChart';

import * as styles from './styles';

export type RectConfig = Record<Regions, IRectDimensions>;

interface Props {
  aggregatedData: IAggregatedGlobalDrug[];
  getOnRegionClickHandler: (
    globalDrug: IAggregatedGlobalDrug
  ) => (event: React.MouseEvent<HTMLDivElement>) => void;
  rectConfig: RectConfig;
}

export const RegionCharts = ({
  aggregatedData,
  getOnRegionClickHandler,
  rectConfig,
}: Props) => (
  <div>
    {aggregatedData.map((drug) => {
      const rect = rectConfig[drug.regionId];

      if (!rect || drug.aggregatedTotal === 0) {
        return null;
      }

      const { top, left } = rectConfig[drug.regionId];

      const style = css({
        top: top - WORLD_MAP_PIE_CHART_SIZE,
        left,
        position: 'absolute',
      });

      return (
        // eslint-disable-next-line max-len
        <div
          key={drug.regionId}
          onClick={getOnRegionClickHandler(drug)}
          css={[styles.container, style]}
        >
          <DrugsPieChart
            key={`${drug.regionId}`}
            susceptibleQuantity={drug.aggregatedSusceptible}
            intermediateQuantity={drug.aggregatedIntermediate}
            resistantQuantity={drug.aggregatedResistant}
            totalSamples={drug.aggregatedTotal}
          />
        </div>
      );
    })}
  </div>
);

export default RegionCharts;
