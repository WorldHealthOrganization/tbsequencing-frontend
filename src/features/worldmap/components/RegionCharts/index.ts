import RegionCharts from './RegionCharts';

export default RegionCharts;
