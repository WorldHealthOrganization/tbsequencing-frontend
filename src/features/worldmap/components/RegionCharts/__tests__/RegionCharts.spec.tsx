import React from 'react';
import { getTreeJSONWithRedux, render } from '../../../../../utils/testUtils/jestUtils';
import RegionCharts from '../RegionCharts';
import { IAggregatedGlobalDrug } from '../../../../../services/drugsApi/models';
import { Regions } from '../../../../drugsView/models';

describe('RegionCharts component test', () => {
  const mockData: IAggregatedGlobalDrug[] = [
    {
      regionId: Regions.EASTERN_MEDITERRANEAN,
      aggregatedIntermediate:
        10,
      aggregatedIntermediateRatioFormatted: 'mock',
      aggregatedResistant:
        10,
      aggregatedResistantRatioFormatted: 'mock',
      aggregatedResistantRatio: 10,
      aggregatedSusceptibleRatioFormatted: 'mock',
      aggregatedSusceptibleRatio: 10,
      aggregatedSusceptible: 10,
      aggregatedIntermediateRatio: 10,
      aggregatedTotal: 1000,
      regionName: 'mockName',
    },
  ];

  const mockConfigPosition = {
    top: 0,
    left: 0,
  };

  const mockConfig = {
    [Regions.EASTERN_MEDITERRANEAN]: mockConfigPosition,
  };

  const mockHandler = (_: IAggregatedGlobalDrug) => jest.fn();

  test('renders RegionCharts component without error', () => {
    render(
      <RegionCharts
        // @ts-ignore
        rectConfig={mockConfig}
        aggregatedData={mockData}
        getOnRegionClickHandler={mockHandler}
      />,
    );
  });

  it('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      <RegionCharts
        // @ts-ignore
        rectConfig={mockConfig}
        aggregatedData={mockData}
        getOnRegionClickHandler={mockHandler}
      />,
    );

    expect(tree).toMatchSnapshot();
  });
});
