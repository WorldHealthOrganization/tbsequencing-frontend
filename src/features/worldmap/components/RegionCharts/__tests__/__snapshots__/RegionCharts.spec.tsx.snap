// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegionCharts component test Should render correctly 1`] = `
<div>
  <div
    className="css-3d67wz-RegionCharts"
    onClick={[MockFunction]}
  >
    <div
      className="recharts-wrapper css-t9zoz8-DrugsPieChart"
      role="region"
      style={
        Object {
          "cursor": "default",
          "height": 98,
          "position": "relative",
          "width": 98,
        }
      }
    >
      <svg
        className="recharts-surface"
        cx="50%"
        cy="50%"
        height={98}
        viewBox="0 0 98 98"
        width={98}
      >
        <title />
        <desc />
        <defs>
          <clipPath
            id="recharts3-clip"
          >
            <rect
              height={88}
              width={88}
              x={5}
              y={5}
            />
          </clipPath>
        </defs>
        <g
          className="recharts-layer recharts-pie"
          tabIndex={0}
        >
          <g
            className="recharts-layer"
          >
            <g
              className="recharts-layer recharts-pie-sector"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              tabIndex={-1}
            />
            <g
              className="recharts-layer recharts-pie-sector"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              tabIndex={-1}
            />
            <g
              className="recharts-layer recharts-pie-sector"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              tabIndex={-1}
            />
          </g>
          <text
            className="recharts-text recharts-label"
            fill="#808080"
            offset={5}
            style={
              Object {
                "fontFamily": "SourceSansBold",
                "fontSize": 16,
              }
            }
            textAnchor="middle"
            width={25}
            x={49}
            y={49}
          >
            <tspan
              dy="0.355em"
              x={49}
            >
              1K
            </tspan>
          </text>
        </g>
      </svg>
    </div>
  </div>
</div>
`;
